### ISSUE TITLE
Insufficient overflow checking with 'sum' type

### ISSUE THREAD

**Issue by ikkeps** [band115/ecirca#33]
With 'sum' type, current overflow checking is wrong if resulting value is greater than any of two values, _and_ greater than MAX_VAL.
Could be fixes with: 

``` diff
diff --git a/priv/ecirca.c b/priv/ecirca.c
index 69d2127..b2a26d5 100644
--- a/priv/ecirca.c
+++ b/priv/ecirca.c
@@ -696,7 +696,7 @@ update_value(ErlNifEnv* env, circactx* ctx,
             break;
         case ecirca_sum:
             sum = val + ctx->circa[idx];
-            if (sum >= val && sum >= ctx->circa[idx]) { /* no overflow */
+            if (sum >= val && sum >= ctx->circa[idx] && sum < MAX_VAL) { /* no overflow */
                 ctx->circa[idx] = sum;
             } else {
                 *ret = TUPLE2(ATOM_ERROR, ATOM_OVERFLOW);
```

**Comment by si14** [comment]
Now you can commit to this repo. But remember: with great power comes great responsibility!

**Comment by ikkeps** [comment]
Thanks. Fixed in 17ff2438c4d9850419ea184495c42203c5b6f9b0.
